File: Doodream/16917 양념 반 후라이드 반.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().split("\n");

function solution() {
  let answer = 0;
  const inputs = input[0].split(" ").map(Number);
  let [a, b, c, x, y] = inputs;
  if (c < (a + b) / 2) {
    const cCount = x < y ? x * 2 : y * 2;
    answer = Math.min(
      a * (x - cCount / 2) + b * (y - cCount / 2) + c * cCount,
      c * Math.max(x, y) * 2
    );
  } else {
    answer = a * x + b * y;
  }
  return answer;
}

console.log(solution());

File: yeonkyung/16917. 양념 반 후라이드 반.py
A, B, C, X, Y = map(int, input().split())

price = A * X + B * Y
min_price = 0
if X <= Y:
    min_price += min(B, C * 2) * (Y - X)
    min_price += C * 2 * X
else:
    min_price += min(A, C * 2) * (X - Y)
    min_price += C * 2 * Y

print(min(price, min_price))
